src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -259,11 +258,12 @@ private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher
             }
         }
     }
-
-    public String constructCommandForTest(String inputArguments) {
-        final String matlabFunctionName =
-                FilenameUtils.removeExtension(MatlabBuilderConstants.MATLAB_TESTS_RUNNER_TARGET_FILE);
-        final String runCommand = "exit(" + matlabFunctionName + "(" + inputArguments + "))";
+    
+    public String constructCommandForTest(String inputArguments, FilePath scriptPath) {
+        final String matlabFunctionName = MatlabBuilderConstants.MATLAB_TEST_RUNNER_FILE_PREFIX
+                + scriptPath.getBaseName().replaceAll("-", "_");
+        final String runCommand = "addpath(genpath('" + scriptPath.getRemote() + "')); "

```

Comment:
Don't forget any time you're injecting a string value into MATLAB code, you have to escape single quotes.

Reply:
Done.